examples/step-31/step-31.cc

```diff
@@ -1040,15 +1054,17 @@ namespace Step31
     // and $\mathbf u^{n-2}$, as well as for the temperatures $T^{n}$,
     // $T^{n-1}$ and $T^{n-2}$ (required for time stepping) and all the system
     // right hand sides to their correct sizes and block structure:
-    stokes_solution.reinit (stokes_block_sizes);
-    old_stokes_solution.reinit (stokes_block_sizes);
-    stokes_rhs.reinit (stokes_block_sizes);
+    IndexSet temperature_partitioning (n_T);
+    temperature_partitioning.add_range(0,n_T);
+    stokes_solution.reinit (stokes_block_sizes, MPI_COMM_WORLD);
```

Comment:
I think this business of generating complete index sets to initialize vectors is not very elegant. What would you think of keeping the original variables that store the sizes of the objects, and then have a free function
```
  IndexSet complete_index_set (const unsigned int size) {
    IndexSet is (size);
    is.add_range (0, size);
    return is;
  }
```
that one could simply call in-place in all of these initialization statements?